rest/src/main/java/com/netflix/conductor/rest/rbac/RbacAccessAspect.java

```diff
@@ -0,0 +1,281 @@
+import java.util.Arrays;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Pointcut;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.HttpClientErrorException;
+
+import com.netflix.conductor.common.metadata.BaseDef;
+import com.netflix.conductor.common.metadata.workflow.StartWorkflowRequest;
+import com.netflix.conductor.common.model.BulkResponse;
+import com.netflix.conductor.core.exception.NotFoundException;
+
+@Aspect
+@Component
+public class RbacAccessAspect {
+
+    private final RbacDbHandler handler;
+
+    private final RbacHttpFilter filter;
+
+    private Object[] arguments;
+
+    public RbacAccessAspect(RbacDbHandler handler, RbacHttpFilter filter) {
+        this.handler = handler;
+        this.filter = filter;
+    }
+
+    @Pointcut("@annotation(com.netflix.conductor.rest.rbac.annotations.RbacAdminAccess)")
+    public void adminAccessMethods() {}
+
+    @Pointcut("@annotation(com.netflix.conductor.rest.rbac.annotations.RbacAdminAccessVoid)")
+    public void adminAccessVoidMethods() {}
+
+    @Pointcut("@annotation(com.netflix.conductor.rest.rbac.annotations.RbacBulkAccess)")
+    public void rbacBulkAccess() {}
+
+    @Pointcut("@annotation(com.netflix.conductor.rest.rbac.annotations.RbacPathVarAccess)")
+    public void rbacPathVarAccess() {}
+
+    /**
+     * Executes methods with admin access control.
+     *
+     * <p>Proceeds only in situations when user is admin. Otherwise, throws 401.
+     *
+     * @param joinPoint The proceeding join point, enabling interception of the method call.
+     * @return The result of the intercepted method call.
+     * @throws Throwable Throws any throwable exception that occurs during method execution.
+     */
+    @Around("adminAccessMethods()")
+    public Object triggerAdminMethods(ProceedingJoinPoint joinPoint) throws Throwable {
+        if (filter.getUser().isAdmin()) {
+            return joinPoint.proceed();
+        } else {
+            throw new HttpClientErrorException(HttpStatus.UNAUTHORIZED);
+        }
+    }
+
+    /**
+     * Executes void methods with admin access control.
+     *
+     * <p>Proceeds only in situations when user is admin. Otherwise, throws 401.
+     *
+     * @param joinPoint The proceeding join point, enabling interception of the method call.
+     * @throws Throwable Throws any throwable exception that occurs during method execution.
+     */
+    @Around("adminAccessVoidMethods()")
+    public void triggerAdminVoidMethods(ProceedingJoinPoint joinPoint) throws Throwable {
+        if (filter.getUser().isAdmin()) {
+            joinPoint.proceed();
+        } else {
+            throw new HttpClientErrorException(HttpStatus.UNAUTHORIZED);
+        }
+    }
```

Comment:
those methods can be merged into 1 with Object return type

Reply:
Removed. All admin endpoints annotated with @AdminRbacAccess.